contracts/erc721/PaymentObligation.sol

```diff
@@ -65,93 +60,41 @@ contract PaymentObligation is UserMintableERC721 {
    * @param _anchorId bytes32 The ID of the document as identified
    * by the set up anchorRegistry.
    * @param _merkleRoot bytes32 The root hash of the merkle proof/doc
-   * @param _collaboratorField string The values of the collaborator leaf
    * It needs to start with a collaborator prefix, ex: collaborator[0]
-   * @param _values string[5] The values of the leafs that is being proved
-   * precise-proofs library.
-   * @param _salts bytes32[5] The salts for the field that is being proved
+   * @param _values string[] The values of the leafs that are being proved
+   * using precise-proofs
+   * @param _salts bytes32[] The salts for the field that is being proved
    * Will be concatenated for proof verification as outlined in
    * precise-proofs library.
-   * @param _proofs bytes32[][5] Documents proofs that are needed
+   * @param _proofs bytes32[][] Documents proofs that are needed
    * for proof verification as outlined in precise-proofs library.
+   * @param _signature signed message used for on chain ACL
    */
   function mint(
     address _to,
     uint256 _tokenId,
     string _tokenURI,
     uint256 _anchorId,
     bytes32 _merkleRoot,
-    string _collaboratorField,
-    string[5] _values,
-    bytes32[5] _salts,
-    bytes32[][5] _proofs
+    string[] _values,
+    bytes32[] _salts,
+    bytes32[][] _proofs,
+    bytes _signature
```

Comment:
I think we should refactor our identity to meta transactions together with a proxy functionality. 

Currently, the `AnchorRepository` and `PaymentObligation` have to implement the signature validation.  It should only happen once inside the identity contract.

In the future a `msg.sender` check for `AnchorRepository` and `PaymentObligation` would be enough.

Reply:
I do not think msg.sender works for a delegate pattern. And in the case of a meta transaction isn't the msg.sender the relayed, proxy contract and you need the signature in order to retrieve the public key?